jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -191,6 +191,12 @@ dereference_exprt checked_dereference(const exprt &expr)
   return result;
 }
 
+dereference_exprt
+element_at_pointer_array_index(const exprt &pointer, const exprt &index)
+{
+  return dereference_exprt{plus_exprt{pointer, index}};
```

Comment:
:bulb: Would it be a good idea to have something like `INVARIANT(can_cast_type<pointer_typet>(pointer.type()));` at the beginning of this function?

Reply:
Done (using `PRECONDITION`).